bson/src/main/org/bson/json/Base64.java

```diff
@@ -128,10 +140,14 @@ public String encode(final byte[] in) {
                 break;
         }
 
-        try {
-            return new String(buffer, "UTF-8");
-        } catch (UnsupportedEncodingException e) {
-            throw new MongoException("UTF-8 Charset is not available");
-        }
+        return byteArrayToString(buffer);
+    }
+
+    @SuppressWarnings("deprecation")
+    private static String byteArrayToString(final byte[] buffer) {
+        return new String(buffer, 0, 0, buffer.length);
```

Comment:
Just a way I found to avoid the UnsupportedEncodingException...this constructor assumes ASCII text.